### PR TITLE
Core components + Lerna integrated 

### PR DESCRIPTION
**Pull request by alebinson**
Shared components repo became a mono-repo which contains react-core and react-components(ACTIVE packages) as well contains STENCIL (not ACTIVE packages).
In addition contains demo apps for convenient react and stencil development(waits for story-book)

### CODE REVIEW

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 
 The repo uses [Yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/) to link dependencies, and to enable running `yarn install` once.
 
 **You should use yarn instead of npm.**
 
 
 ## Projects
-* **mc-conflict-ui** - UI app for resolving conflicts.
-* **mc-react-components** - Component library for shared components.
+* **conflict-ui** - UI app as a demo app.
```
[the remark this answers is not in the document]
Comment (alebinson) [reply README.md]: done

File: README.md [hunk 2]
```diff
@@ -25,9 +31,37 @@ The repo uses [Yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/)
...
 }
 ```
 5. Add any external dependencies you would like using yarn.
 6. start hacking :wink:
+
+## Build
+1. Be logged in to NPM
+```json
+  npm login
+```
```
[the remark this answers is not in the document]
Comment (alebinson) [reply README.md]: Nice one

File: lerna.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "lerna": "2.11.0",
+    "packages": [
+      "packages/react-components",
+      "packages/react-ui-components"
+    ],
+    "version": "independent",
```
[the remark this answers is not in the document]
Comment (alebinson) [reply lerna.json:7]: Still not sure, let's stick **independent** for awhile

File: packages/react-components/src/lib/box/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { McBox as Box } from './box';
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/box/index.ts]: Fixed by running **prettier** task on all source code

File: packages/react-components/src/lib/cssbaseline/cssbaseline.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+export const McScopedCssBaseline: React.FC<ScopedCssBaselineProps> = (props: ScopedCssBaselineProps) => 
+{
+    const {children, ...rest} = props;
+    return (
+        <ScopedCssBaseline {...rest}>
+            {children}
+        </ScopedCssBaseline>
+    );
+}
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/cssbaseline/cssbaseline.tsx]: Fixed by running **prettier** task on all source code

File: packages/react-components/src/lib/date-range-picker/date-range-picker.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+import { isValid, isBefore } from 'date-fns';
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDateTimePicker,
+} from '@material-ui/pickers';
+import DateFnsUtils from '@date-io/date-fns';
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import { ThemeProvider } from '@material-ui/core';
+
+import { Button } from '@map-colonies/react-core/dist';
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/date-range-picker/date-range-picker.tsx]: **dist** removed

File: packages/react-components/src/lib/index.ts [hunk 7]
```diff
@@ -3,3 +3,7 @@ export * from './map-filter-container';
 export * from './models';
 export * from './ol-map';
 export * from './smart-table';
+export * from './popover';
+export * from './box';
+export * from './cssbaseline';
+export * from './theme';
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/index.ts:9]: Fixed by running **prettier** task on all source code

File: packages/react-components/src/lib/map-filter-container/polygon-selection-ui.tsx [hunk 8]
```diff
@@ -1,13 +1,26 @@
 import React from 'react';
-import { Menu, MenuItem, Button } from '@material-ui/core';
+// import { Menu, MenuItem, Button } from '@material-ui/core';
+// import Tooltip from '@material-ui/core/Tooltip';
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/map-filter-container/polygon-selection-ui.tsx]: done

File: packages/react-components/src/lib/popover/index.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+// import {McPopover as Popover } from './popover';
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/popover/index.ts]: done

File: packages/react-components/src/lib/theme/theme.ts [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+--mdc-theme-text-primary-on-light: rgba(0, 0, 0, 0.87);
+--mdc-theme-text-secondary-on-light: rgba(0, 0, 0, 0.54);
+--mdc-theme-text-hint-on-light: rgba(0, 0, 0, 0.38);
+--mdc-theme-text-disabled-on-light: rgba(0, 0, 0, 0.38);
+--mdc-theme-text-icon-on-light: rgba(0, 0, 0, 0.38);
+--mdc-theme-text-primary-on-dark: white;
+--mdc-theme-text-secondary-on-dark: rgba(255, 255, 255, 0.7);
+--mdc-theme-text-hint-on-dark: rgba(255, 255, 255, 0.5);
+--mdc-theme-text-disabled-on-dark: rgba(255, 255, 255, 0.5);
+--mdc-theme-text-icon-on-dark: rgba(255, 255, 255, 0.5); */
```
[the remark this answers is not in the document]
Comment (alebinson) [reply packages/react-components/src/lib/theme/theme.ts]: Yep, on purpose

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**